main/src/library/Result.flix

```diff
@@ -409,4 +409,18 @@ namespace Result {
         case Err(_) => Iterator.empty()
         case Ok(x)  => Iterator.singleton(x)
     }
+
+    ///
+    /// Returns `Ok(x)` if `f` was invoked without throwing an exception.
+    /// Return `Err(e)`, if `f` throws a Java `RuntimeException`,
+    /// where `e` is the error message.
+    ///
+    pub def try(f: Unit -> a & ef): Result[a, String] & Impure =
+        try {
+            Ok(f())
+        } catch {
+            case e: ##java.lang.RuntimeException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(e))
```

Comment:
by casting this to pure

Reply:
Is that safe?

Reply:
For getMessage, yes.

Reply:
In the try body you have a call to f which has effect `ef` and thats the effect that should be there.